src/main/java/io/aiven/kafka/connect/opensearch/spi/ClientsConfiguratorProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 Aiven Oy
+ * Copyright 2016 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.opensearch.spi;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.ServiceLoader;
+
+import io.aiven.kafka.connect.opensearch.OpensearchSinkConnectorConfig;
+
+public final class ClientsConfiguratorProvider {
+    private ClientsConfiguratorProvider() {
+    }
+
+    /**
+     * Use {@link ServiceLoader} mechanism to discover available configurators for Opensearch and Kafka clients 
```

Comment:
Nit:
```suggestion
     * Use {@link ServiceLoader} mechanism to discover available configurators for Opensearch clients 
```
(Kafka clients use JAAS; we probably won't need to implement our own discovery logic for them)

Reply:
:+1: I just kept is as an option, will state it explicitly

Reply:
I think we should remove this completely; there's very little chance that we'll want to add our own discoverability logic for Kafka clients, especially since there's no known mechanism to modify them because there's no API for connectors to access their own Kafka clients.